lib/rouster.rb

```diff
@@ -212,7 +212,15 @@ def initialize(opts = nil)
           @logger.debug(':instance specified, will connect to existing OpenStack instance')
           inst_details = self.ostack_describe_instance(@passthrough[:instance])
           raise ArgumentError.new(sprintf('No such instance found in OpenStack - %s', @passthrough[:instance])) if inst_details.nil?
-          @passthrough[:host] = inst_details.addresses["NextGen"][0]["addr"]
+          if defined?(inst_details.addresses["NextGen"][0]["addr"])
+            @passthrough[:host] = inst_details.addresses["NextGen"][0]["addr"]
+          else
+            inst_details.addresses.each_key do |address_key|
+              if defined?(inst_details.addresses[address_key][0]['addr'])
+                @passthrough[:host] = inst_details.addresses[address_key][0]['addr']
```

Comment:
I agree with that and that would not change the result even if addresses contains key 'NextGen'. I will also update the logic here to make it just do a general iterate:
https://github.com/chorankates/rouster/blob/master/plugins/openstack.rb#L43